src/Faker/Container/Container.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Faker\Container;
+
+use Psr\Container\ContainerInterface;
+
+/**
+ * A simple implementation of a container.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ * @experimental This class is experimental and does not fall under our BC promise
+ */
+final class Container implements ContainerInterface
+{
+    /**
+     * @var array
+     */
+    private $container = [];
+    private $cache = [];
+
+    public function __construct(array $container)
+    {
+        $this->container = $container;
+    }
+
+    public function get($id)
+    {
+        if (isset($this->cache[$id])) {
```

Comment:
Should we assert here that `$id` is a `string`?

Reply:
Yeah, makes sense